core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2272,6 +2273,25 @@ private[spark] object Utils extends Logging {
     }.map(threadInfoToThreadStackTrace)
   }
 
+  /** Return a heap dump */
+  def getHeapDump: InputStream = {
```

Comment:
I'm not sure whether this is a safe behavior, a large HeapDump may make the `/tmp` no free space

Reply:
It will be deleted after each dump